sympy/stats/drv.py

```diff
@@ -157,3 +164,44 @@ def compute_characteristic_function(self, expr, **kwargs):
             return self.distribution.compute_characteristic_function(**kwargs)
         else:
             raise NotImplementedError()
+
+    def restricted_domain(self, condition):
+        rvs = random_symbols(condition)
+        assert all(r.symbol in self.symbols for r in rvs)
+        if (len(rvs) > 1):
+            raise NotImplementedError(filldedent('''Multivariate discrete
+            random variables are not yet supported.'''))
+        conditional_domain = reduce_rational_inequalities_wrap(condition,
+            rvs[0])
+        conditional_domain = conditional_domain.intersect(self.domain.set)
+        return conditional_domain
+
+    def probability(self, condition):
+        _domain = self.restricted_domain(condition)
+        if condition is False or _domain is S.EmptySet:
+            return S.Zero
+        if condition is True or _domain == self.set:
+            return S.One
+        try:
+            return self.eval_prob(_domain)
+        except:
```

Comment:
Do not use bare except. This will mask any error in eval_prob. Is there any reason eval_prob cannot raise NotImplementedError on its own? This also should check for a None return type and raise NotImplementedError then.

Reply:
Actually shouldn't it just return an unevaluated object?